src/battle/battle.cpp

```diff
@@ -89,20 +89,21 @@ void Battle(MainCharacter& main_character, std::vector<Creature>& foes)
 				std::vector<std::string> main_character_attack_options(options_size, "");
 				for (size_t iter = 0; iter < foes_ptrs.size(); iter++)
 				{
-					main_character_attack_options[iter] =
-						foes_ptrs[iter]->get_display_name();
+					main_character_attack_options[iter] = utils::UppercasedFirstChar(
+						foes_ptrs[iter]->get_display_name());
 				}
 				user_choice = utils::GetUserConstrainedChoice(main_character_attack_options);
 
 				MainCharacterAttack(*main_character_ptr, *(foes_ptrs[user_choice - 1]));
 
 				// Eliminate (erase) foes with 0 hp
-				if (foes_ptrs[user_choice - 1]->get_hp() == 0)
+				if (foes_ptrs[foes_ptrs.size() - 1]->get_hp() == 0)
```

Comment:
This was fine. Only the `j = foes_ptrs.size()` inside the for loop further down needed to stay the same.